src/kernel/syscall.rs

```diff
@@ -341,7 +341,13 @@ impl SysCalls {
         #[cfg(all(target_os = "none", feature = "kernel"))]
         {
             let pid = argraw(0);
-            kill(pid)
+
+            if pid == 0 {
+                println!("Operation not permitted: Kill root process");
```

Comment:
Thank you for the fantastic update! I appreciate your efforts. I tested this branch by killing the root process, and while doing so, I noticed that the error message output from the Kernel via println is a bit verbose. I think we might want to remove it for now. Ideally, we'd like to keep the Kernel's message output to a minimum until we implement a logging mechanism. Since the Kernel is already communicating with userland through Err(PermissionDenied), it seems redundant to have the println output as well. Could you please comment it out for now? Thanks again for your repeated efforts!

Reply:
@o8vm  Perfect! Thank you for your support and to recognize my efforts and willing to contribute. :)